119 create docs for our project (#126)
ZPP-MURMURAS/ZPP_Murmuras: docs/terminology_doc.md

# Terminology used in our project
This document provides a list of terms used in our project and their definitions. 

___ 
## Model naming conventions
Our project uses a specific naming convention for models. The naming convention is as follows:

>     bert-{selection / extraction}-{data set: json / plain}-{fine tune type}

Reviewer: ```suggestion
    bert-{selection / extraction}-{data set: json / pl}-{fine tune type}
```
To make things shorter

>     bert-{selection / extraction}-{data set: json / plain}-{fine tune type}

Reviewer: Please elaborate on the fine tune type

    llama-{fine tune type}

Selection refers to coupon selection, ie. identifying and selecting the coupons from the text. Extraction refers to field extraction, ie. extracting the fields from the coupons such as product name, discount, etc.

## Data set naming conventions
#### Llama data sets

> We have four types of data sets for llama models:

Reviewer: ```suggestion
We have four types of dataset types for llama models:
```

Author: I'm not sure this makes sense though. "We have four [types] of dataset [types] for..." - we are using "types" twice

Reviewer: hmm, fair. Then please change:
data sets -> datasets

Author: done

    1. one_input_one_output_wrequest
    2. one_input_one_output_wthrequest
    3. one_input_multiple_outputs_wrequest
    4. one_input_multiple_outputs_wthrequest

We are using the last two dataset types in our project.

> In the naming convention, the following abbreviations are used:

Reviewer: ```suggestion
llama-ds-w - dataset with one_input_multiple_outputs_wrequest
llama-ds-wth - dataset with one_input_multiple_outputs_wthrequest
```

That's all we need to know. We don't need fancy language for that.

    w - meaning with requests to llama
    wth - meaning without requests to llama

Henceforth, **llama-ds-w** refers solely to one_input_multiple_outputs_wrequest, while **llama-ds-wth** refers solely to one_input_multiple_outputs_wthrequest.

#### BERT data sets
We are using two types of data sets for bert models:

    1. json (.json)

>     2. plain text (.pl)

Reviewer: .pl and .json suggest file extentions.

BERT data sets are named as follows:

    **bert-{selection / extraction}-ds.{json / pl}**

___
## CSV file formats
We have two types of CSV file formats:

    format=1: values in "content_full" column are in the form of a list of strings without quotation marks separated by ", " (commas followed by a space)

    format=2: values in "content_full" column are in the form of a list of strings surrounded by single quotation marks and separated by "," (commas without spaces after them)

___
## Formats
### Coupon formats

> We have two coupon formats representing the data associated with a signle coupon. We have two levels of complexity for coupons: simple and normal/extended.

Reviewer: ```suggestion
We have two coupon formats representing the data associated with a single coupon. We have two levels of complexity for coupons: simple and normal/extended.
```

```python
@dataclass()

> class Coupon:

Reviewer: @nawiju @SzymonKozl @kdybek Weren't this format deprecated? I'm aware that benchmark accepts two different formats, but I thought that this one was removed long time ago.

The benchmark receives a .csv file containing the ground truth, ie. the coupons extracted from the input data. This file must contain the following headers if it will process simple coupons
```txt
'product_name', 'discount_text', 'valid_until', 'activation_text'

```
or the following headers if it will process extended coupons
```txt
'product_name', 'new_price', 'old_price', 'percents', 'other_discounts', 'dates'
```

The benchmark outputs both the similarity scores and identifies any isolated/"lonely" coupons, coupons that are not associated with any other coupons in the dataset.

#### Pipelines

Each pipeline receives as input a .csv file containing the following headers: "view_depth", "text", "description", "seen_timestamp", and "is_visible". 

It will output the identified coupons in a .csv file that is of the same format as the ground truth file described above.

> #### BERT models

Reviewer: Maybe you could group the information by models? It's the nth time I see BERT related stuff. It's not very readable, I think that a better approach would be to have BERT and LLAMA related sections

Reviewer: Hmm, fair; let's discuss this on Thursday.
